pennylane/grouping/utils.py

```diff
@@ -658,6 +658,38 @@ def is_qwc(pauli_vec_1, pauli_vec_2):
     return True
 
 
+def are_pauli_words_qwc(lst_pauli_words):
+    """Given a list of observables assumed to be valid Pauli words, determine if they
+     are all mutually qubit-wise commuting.
+
+    This implementation has time complexity ~ O(m * n) for n wires and m Pauli words.
+
+    Args:
+        lst_pauliwords (list[Observable]): list of observables (assumed to be valid Pauli words).
+
+    Returns:
+        (bool): True if they are all qubit-wise commuting, false otherwise.
+    """
+    wire_dict = {}
+
+    for op in lst_pauli_words:  # iterate over the list of  m observables
```

Comment:
(Minor): m seems a bit confusing when just looking at the code.
```suggestion
    for op in lst_pauli_words:  # iterate over the list of observables
```